src/Language/Mimsa/Interpreter/BuiltIns.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Language.Mimsa.Interpreter.BuiltIns where
+
+import Data.Text (Text)
+import qualified Data.Text as T
+import Language.Mimsa.Types
+
+wrappedName :: Variable -> Variable
+wrappedName = appendToVar "__unwrapped"
+
+appendToVar :: Text -> Variable -> Variable
+appendToVar tx (BuiltIn (Name n)) = BuiltIn (Name (n <> tx))
+appendToVar _ a = a
+
+wrappedVarName :: Variable -> Int -> Variable
```

Comment:
This is pretty broken because it doesn't allow the same function to be partially applied twice without sharing the variable. Needs to be rewritten in the `App` monad so we can check what's already here and ensure uniqueness.